src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -447,6 +447,163 @@ JSON example:
 }
 ```
 
+## Arterial Spin Labeling perfusion data
+
+Template:
+
+```Text
+sub-<label>/[ses-<label>/]
+    perf/
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.nii[.gz]
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.json
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_aslcontext.tsv
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.nii[.gz]
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.json
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_labeling.jpg
+```
+
+The complete ASL time series should be stored as a 4D NIfTI file in the original acquisition order, accompanied by two ancillary 
+files: `*_asl.json` and `*_aslcontext.tsv`. The `*_asl.nii.gz` and `*_m0scan.nii.gz` should contain appropriately scaled data and no scaling factors 
+should be stored in the ancillary files or in the NIfTI header. 
```

Comment:
Sorry if I missed if and where this was discussed: what is the rationale for banning scaling factors in NIfTI headers? Does it mean `scl_slope` can only be `1.0`? From my naive viewpoint, it seems to be prescriptive of a level that should be abstracted away.

Reply:
Hi @gllmflndn, this was indeed already discussed before, so I will copy @HenkMutsaerts 's response here. Feel free to ask your remaining questions:

ASL BIDS certainly takes the scale factors in account. We considered two options, of which we chose the second, as it has less potential for confusion.

1. keep image values as is, retain the scale factors, such that scaling needs to be dealt with in post processing
2. use the scale factors to scale the image values, removing the need for scaling in post processing.

We chose that the ASL BIDS standard should not have scale factors, which puts the responsability for correct scaling to the DICOM2BIDS conversion stage, when any incorrect scaling can still be dealt with, leaving no confusion as to whether scaling has or has not been applied in the BIDS data.

Hope this explains it!

Reply:
Thank you @patsycle for letting me know and sorry for not having seen this response before.

I am still a bit confused though - if this restriction remains, I think it should be clarified what `"appropriately scaled data and no scaling factors should be stored in the ancillary files or in the NIfTI header"` exactly means. If this means `scl_slope` and `scl_inter`  in the NIfTI header have to be `1` and `0` respectively then it should be explicit. A DICOM to NIfTI converter would rarely change the data type while it means here that `datatype` will almost always be `float` or `double`.
Why is there a mention of `ancillary files` regarding scaling as there doesn't seem to be any scale-related parameters anyway?

And apologies if I misunderstand something or if it's too late to have this conversation but I don't see what is specific with ASL data that would require a restriction on the scaling factors in the NIfTI header.

Reply:
I am linking this question to @HenkMutsaerts , who will probably be able to explain this much better than me!

Reply:
@gllmflndn I also am not fond of this restriction. If I understood the earlier conversation, the reason that scale factors are complicated is that the DICOMs can have per-volume scale factors, and this is an issue with sequences where the M0, CBF and/or deltaM volumes are in the same series as the label/control volumes. To create a NIfTI with a shared slope/intercept, at least some of the volumes would have to be rescaled, introducing rounding error.

My personal preference would be to separate the label/control volumes from the M0/CBF/deltaM volumes during conversion, but I don't know how practicable that is.

(This is a rehash of https://github.com/bids-standard/bids-specification/pull/652#issuecomment-721488288, btw.)

Reply:
@effigies Although I agree this might seems cleaner, I am afraid it is not practicle. 

The splitting up or not has been discussed many times with many ASL experts and all agreed on keeping the timeseries as acquired. The main disadvantage of ASL is that the way the data is stored (both the aslcontext within the timeseries, as scaling) is very much different between vendors, scanners, sequences, which is why flexibility is needed. 

I dug up the meeting notes on this discussion and these were the 2 main reasons why we chose for keeping the dataset in the original acquisition order:

- this allows reviewing temporal effects such as head motion (for which perfusion is very sensitive due to the low SNR)

- it allows flexibility which will be needed for more complex asl-approaches and labeling patterns. The current extension is focussing on these approaches which were included in the recommendations paper (Alsop 2015). However, more advanced approaches already exist and will probably gain more interest. Future updates of the ASL extensions might be needed and revisiting the basics of the ASL extension due to 'not flexible enough' would be inefficient.

I hope this helps in the discussion.

Reply:
My personal preference would be the same than @effigies but if it has been thoroughly discussed and decided against, fine :smile: Despite this, I still don't see why there should be this statement in the specs: `and no scaling factors should be stored in the ancillary files or in the NIfTI header`. Once all of the data have been concatenated in a 4D file, in one way or another, the NIfTI scaling and data types could be anything and I don't see the need to be prescriptive here.

Again, apologies if I miss something obvious.

Reply:
@patsycle Thanks for this explanation. That makes much more sense now.

I think I agree with @gllmflndn about this specific line, then. Whether to rescale or not feels like a decision that conversion software can handle without being dictated by BIDS.

Reply:
Sorry, for this understandable confusion. Appropriate analysis of ASL data is by design dependent on absolute quantification and having applied the correct scale factors (where this may not be an issue with other scan types). To avoid confusion if additional scale slopes - such as the MRScaleSlope in Philips, or the arbitrary factors that other vendors use - were or were not already applied, it seems most useful to put the correct scaling responsibility in the DICOM2BIDS conversion step. When ASL data is in the "BIDS standard" there should be no confusion of scale slopes left, and data should be fully ready for automatic processing.

Indeed, this doesn't mean that the NIfTI file cannot have a scaleslope nor that it has to be in double/floating-point per see, so we could remove this statement. However, that would require us to state something like: "ASL data should be properly scaled and any non-standard scale slopes - including those found in the private fields or not stored in the DICOMs - should be taken into account. Would you prefer this? I'm not sure if this would be confusing. Also, if we make floating point & NIfTI header scale slope "1" obliged, then it is easier to check if scale slopes were taken into account during conversion.

Reply:
@HenkMutsaerts Thanks. I think it's just a matter of rephrasing. From the point of view of BIDS, there are no restrictions on the ASL NIfTI files. Then, you seem to suggest to add a note to DICOM2BIDS developers that data have to be scaled appropriately (potentially taking into account non-standard scale slopes found in DICOM private fields) to accommodate the unique scaling factor NIfTI files can have for all of the contained volumes.

Reply:
Yes, this makes sense! Thanks